### PR TITLE
11 get reviews queries

### PR DESCRIPTION
**Pull request by Daniel17971**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -378,6 +378,73 @@ describe("GET /api/users", () => {
...
 
+describe("GET /api/reviews?query=value", () => {
+  it("200: can select all reviews for a category /api/reviews?category=dexterity", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity")
+      .expect(200)
+      .then((response) => {
+        const reviews = response.body.reviews;
+
+        let flag = true;
```
Comment (Hard-Luck) [inline __tests__/app.test.js:389]: we need to test that the response has a length else the foreach will fail, also look at jest docs, you can have an expect statement in a foreach that will specifically check each element

File: __tests__/app.test.js [hunk 2]
```diff
@@ -378,6 +378,73 @@ describe("GET /api/users", () => {
...
+      .expect(200)
+      .then((response) => {
+        const reviews = response.body.reviews;
+        expect(reviews).toBeSortedBy("created_at", {
+          descending: false,
+          coerce: true,
+        });
+      });
+  });
+  it("400: bad request when query asks for not real category etc", () => {
```
Comment (Hard-Luck) [inline __tests__/app.test.js:422]: would this be a 400 or 404? could the category exist in theory?

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Along with the comments left I would look through the test data and find a category with no reviews and ensure you still get the ok response but with an empty array returned

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
The model implementation is also not very dry I would look to try and build the query string dynamically and then only have to do a final db.query after all your checking and query building is complete

**Comment by Daniel17971** [comment]
May be better as 404 but can't think of a way to differentiate since many words could eventually be valid queries if properties added overtime. 400 best I can do.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -378,6 +378,73 @@ describe("GET /api/users", () => {
...
+    return request(app)
```
Comment (pippaamy) [inline __tests__/app.test.js:423]: this will be a 404

**Review by pippaamy — CHANGES_REQUESTED** [review]
Good job, just some changes to make to these tests!

**Pull request merged**